R/module_visualize_star_system.R

```diff
@@ -0,0 +1,43 @@
+#' Module Visualizing Star Systems
+#' @name module_visualize_star_system
+#' @details
+#' This module provides a dynamically refreshing plot component
+#' for visualizing star systems based on data provided
+NULL
+
+#' @param id A unique identifier for the module.
+#' @returns A Shiny UI object.
+#' @describeIn module_visualize_star_system UI function for the system mapping module.
+#' @export
+visualize_star_system_ui = function(id) {
+  ns = shiny::NS(id)
+  shiny::plotOutput(ns("system_plot"))
+}
+
+#' @param id A unique identifier for the module.
+#' @param plot_data A reactive expression returning a data frame with information about the system.
+#' @param show_hz A reactive boolean expression determining whether to show habitable zone.
+#' @returns A reactive expression containing the selected value(s).
+#' @describeIn module_visualize_star_system server function for the system mapping module.
+#' @export
+visualize_star_system_server = function(id, plot_data, show_hz) {
+  shiny::moduleServer(id, function(input, output, session) {
+    logger::log_trace("{id} rendering star system plot")
+
+    output$system_plot = shiny::renderPlot({
+      shiny::req(plot_data())
+      shiny::req(show_hz)
```

Comment:
issue: You need to require a value from the reactive expression, not the reactive expression itself.
```suggestion
      shiny::req(show_hz())
```

So basically `shiny::req()` checks if the value is _truthy_. A reactive object will always be _truthy_, even if its value is not. Think of it like this:


```R
some_function = function() {
  return(NULL)
}

is.null(some_function) # FALSE, because we are checking the function object
is.null(some_function()) # TRUE, because we are checking the returned value
```